### PR TITLE
Added Rod Cutting Problem [Java]

### LINKED ISSUE #163: Rod cutting problem [Java]
http://www.geeksforgeeks.org/dynamic-programming-set-13-cutting-a-rod/

### PR DESCRIPTION
**Pull request by Prakash2403**
Fixes #163 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: rod_cutting_problem/RodCutting.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Problem Statement: Given a rod of length n inches and an array of prices that contains prices of all pieces of size smaller than n. 
+Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+
+Time Complexity: O(n^2)
+Space Complexity: O(n)
+*/
+
+public class RodCutting{
+ 
+   static int cutRod(int price[],int n){
```
Comment (singhpratyush) [inline rod_cutting_problem/RodCutting.java]: Make it `public`.

File: rod_cutting_problem/RodCutting.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        int best_price[] = new int[n+1];
+        for(int i=1;i<best_price.length;i++)
```
Comment (singhpratyush) [inline rod_cutting_problem/RodCutting.java]: Add space after the semicolons.

File: rod_cutting_problem/RodCutting.java (hunk 1 above)
Comment (singhpratyush) [inline rod_cutting_problem/RodCutting.java]: The function would work only if the first argument if provided.

File: rod_cutting_problem/RodCutting.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            best_price[i] = price[i-1];       
+        for (int i = 1; i<=n; i++)
+            for (int j = i; j < n+1; j++)
+                best_price[j] = Math.max(best_price[j], best_price[j-i] + best_price[i]);
+        return best_price[n];
+    }
+
+    public static void main(String args[]){
```
Comment (Monal5031) [inline rod_cutting_problem/RodCutting.java]: Please add brackets with type descriptor as `String[] args`

File: README.md [hunk 4]
```diff
@@ -32,7 +32,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
 | [Quick Select](https://en.wikipedia.org/wiki/Quickselect) | [\[X\]](quick_select/quick_select.c) | | | | |
 | [Quicksort](https://en.wikipedia.org/wiki/Quicksort) | [\[X\]](quicksort/quicksort.c) | | [\[X\]](quicksort/QuickSort.java) | [\[X\]](quicksort/quick_sort.py) | |
 | [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes) | [\[X\]](sieve_of_eratosthenes/sieveOfEratosthenes.c) | | | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.py) | [\[X\]](sieve_of_eratosthenes/sieve_of_eratosthenes.go) |
-
+| [Rod Cutting Problem](http://www.geeksforgeeks.org/dynamic-programming-set-13-cutting-a-rod/) | | |[\[X\]](rod_cutting_problem/RodCutting.java) | | |
```
Comment (Monal5031) [inline README.md]: Alphabetical ordering of list is followed.

**Comment by Prakash2403** [comment]
@Monal5031 @singhpratyush @aviaryan please review.

File: rod_cutting_problem/RodCutting.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+
+Time Complexity: O(n^2)
+Space Complexity: O(n)
+*/
+
+public class RodCutting{
+ 
+    public static int cutRod(int price[]) {
+       	int n = price.length;
```
Comment (Monal5031) [inline rod_cutting_problem/RodCutting.java]: Inconsistent indentation here, probably due to mixed tabs and spaces

File: rod_cutting_problem/RodCutting.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        int best_price[] = new int[n+1];
```
Comment (Monal5031) [inline rod_cutting_problem/RodCutting.java]: make it `int[]`

File: rod_cutting_problem/RodCutting.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        for(int i=1;i<best_price.length;i++)
```
Comment (Monal5031) [inline rod_cutting_problem/RodCutting.java]: Use consistent spacing around binary operator throughout code.

**Comment by Prakash2403** [comment]
@Monal5031 Please review again.

File: rod_cutting_problem/RodCutting.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+
+Time Complexity: O(n^2)
+Space Complexity: O(n)
+*/
+
+public class RodCutting{
+
+    public static int cutRod(int[] price) {
+       	int n = price.length;
```
Comment (Monal5031) [inline rod_cutting_problem/RodCutting.java]: This inconsistent indentation still remains here.

**Comment by Monal5031** [comment]
@Prakash2403 remove `{}` from your PR description, it should be `Fixed #163`, this links the issue with this PR.

File: rod_cutting_problem/RodCutting.java (hunk 8 above)
Comment (Prakash2403) [inline rod_cutting_problem/RodCutting.java]: It's consistent. 4 spaces for first block, 8 spaces for inside statements.

Comment (aviaryan) [inline rod_cutting_problem/RodCutting.java]: But it is the first block.

File: rod_cutting_problem/RodCutting.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+/**
+Problem Statement: Given a rod of length n inches and an array of prices that contains prices of all pieces of size smaller than n. 
+Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+
+Time Complexity: O(n^2)
+Space Complexity: O(n)
+*/
+
+public class RodCutting{
```
Comment (aviaryan) [inline rod_cutting_problem/RodCutting.java]: Add a space before {

File: rod_cutting_problem/RodCutting.java (hunk 8 above)
Comment (Monal5031) [inline rod_cutting_problem/RodCutting.java]: @Prakash2403 it happens a lot with me too. Just remove all the spaces first, and insert them again.
This usually happens when we hit enter, it assumes tabspace for that part and we add space for our requirements.
Also make sure you remove the spaces for lines below too, use space for all of them.

Comment (Prakash2403) [inline rod_cutting_problem/RodCutting.java]: Ok. Let me try. I haven't faced this issue before. .

**Comment by Prakash2403** [comment]
@Monal5031 Please review it again. Hope it's final review.

**Comment by Monal5031** [comment]
@Prakash2403 Thanks, everything looks good now. 👍

**Comment by singhpratyush** [comment]
Thanks @Prakash2403.

Please set up git properly on your  machine so that Github can recognize your commits.

**Pull request merged**